packages/ve-hemi-actions/actions/wallet/veHemi.ts

```diff
@@ -110,60 +251,18 @@ const runCreateLock = ({
       const veHemiAddress = getVeHemiContractAddress(walletClient.chain!.id)
       const hemiTokenAddress = await memoizedGetHemiTokenAddress(walletClient)
 
-      // Using @ts-expect-error fails to compile so I need to use @ts-ignore
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore because it works on IDE, and when building on its own, but fails when compiling from the portal through next
-      const allowance = await getErc20TokenAllowance(walletClient, {
-        address: hemiTokenAddress,
-        owner: account,
-        spender: veHemiAddress,
+      const approved = await handleApproval({
+        account,
+        amount,
+        approvalAmount,
+        emitter: emitter as EventEmitter<ApprovalEvents>,
```

Comment:
In order to avoid this casting, I think we can do something similar to [this](https://github.com/hemilabs/ui-monorepo/blob/137abbcd0e1fb864338a7222bd492d47ec2f138d/packages/hemi-tunnel-actions/src/utils.ts#L78). Check the usages of `handleWaitDeposit` and see how it automatically infers the correct type of `EventEmitter<T>`

Reply:
Nice! applied here 23656193192048190d45f29d034931e3b5576a8d and the it infers automatically. Thanks!

Reply:
soo... if it infers automatically, can we remove the casting?

The goal should be to be able to just do

```suggestion
        emitter: emitter,
```